crypto/ts/ts_rsp_sign.c

```diff
@@ -719,12 +729,27 @@ static int ts_RESP_sign(TS_RESP_CTX *ctx)
         goto err;
     }
 
-    certs = ctx->flags & TS_ESS_CERT_ID_CHAIN ? ctx->certs : NULL;
-    if ((sc = ess_SIGNING_CERT_new_init(ctx->signer_cert, certs)) == NULL)
-        goto err;
-    if (!ESS_add_signing_cert(si, sc)) {
-        TSerr(TS_F_TS_RESP_SIGN, TS_R_ESS_ADD_SIGNING_CERT_ERROR);
-        goto err;
+    if (ctx->flags & TS_ESS_CERT_ID_V2) {
+        certs = ctx->flags & TS_ESS_CERT_ID_CHAIN ? ctx->certs : NULL;
+        if (!
```

Comment:
break this up into two lines.  sc2 = ... and then if (!sc1 ==  NULL)
dont use ! for pointer tests any more.